rust/core-lib/src/tabs.rs

```diff
@@ -541,7 +541,11 @@ impl CoreState {
     fn do_set_language(&mut self, view_id: ViewId, language_id: LanguageId) {
         if let Some(view) = self.views.get(&view_id) {
             let buffer_id = view.borrow().get_buffer_id();
-            self.config_manager.override_language(buffer_id, language_id);
+            self.make_context(view_id).unwrap().language_changed(&language_id);
```

Comment:
I don't like the fact that `self.make_context(view_id).unwrap()` is repeated, but if I do `let context = self.make_context(view_id).unwrap()`, I cannot compile because I borrow `self` mutably and immutably at the same time...